plugins/inputs/memcached/memcached.go

```diff
@@ -3,26 +3,39 @@ package memcached
 import (
 	"bufio"
 	"bytes"
+	"crypto/tls"
 	"fmt"
 	"net"
 	"strconv"
 	"time"
 
 	"github.com/influxdata/telegraf"
+	tlsint "github.com/influxdata/telegraf/plugins/common/tls"
 	"github.com/influxdata/telegraf/plugins/inputs"
+	"golang.org/x/net/proxy"
```

Comment:
Thanks for the review. Looks like the `common/proxy` package is mostly just a wrapper around `x/net/proxy` to generate a SOCKS5 or HTTP `Dialer` from a Telegraf TOML config. I haven't touched that part of the codebase before but I imagine it's useful to keep around.

Reply:
The reason I'm asking is that I'm a bit hesitant to have both ways in the code-base as it calls for confusion. So ideally you should use `common/proxy` or we should remove `common/proxy` altogether (if it replicates an upstream package). Do you think that's possible?

Reply:
Oh, I see. The usage of `golang.org/x/net/proxy` in this PR is just to type the `dialer` variable against the `proxy.Dialer` interface type. There isn't any actual usage of proxy logic.

If you like, I can restructure the logic so that the `golang.org/x/net/proxy` import is no longer needed. But there would be more code duplication. Something like

```
plaintextDialer := ...
var tlsDialer *tls.Dialer

if enable tls {
  tlsCfg, err := m.ClientConfig.TLSConfig()
  ...
  tlsDialer = ...
}

...

// following repeated twice, for each of unix socket and TCP address endpoints
if enable tls {
  conn, err := tlsDialer.Dial(...)
  ...
} else {
  conn, err := plaintextDialer.Dial(...)
  ...
}

```